Source/WebCore/page/Page.h

```diff
@@ -360,6 +360,8 @@ class Page : public RefCounted<Page>, public Supplementable<Page>, public CanMak
     const URL& mainFrameURL() const { return m_mainFrameURL; }
     SecurityOrigin& mainFrameOrigin() const;
     WEBCORE_EXPORT void setMainFrameURL(const URL&);
+    WEBCORE_EXPORT void setMainFrameURLFragment(String&&);
+    String mainFrameURLFragment() const { return m_mainFrameURLFragment; }
```

Comment:
So text fragment will never be used in an iframe?

Reply:
https://wicg.github.io/scroll-to-text-fragment/#motivation
Yes, they are top level navigation only.